### DISCUSSION TITLE
Different colors for line chart depending on the value

### DISCUSSION THREAD

**Discussion by pachi81** [patrykandpatrick/vico#959] [Questions]
### Question

Hi,
your work looks really great and it seems the perfect library for my app. 
I want show the glucose levels in a line chart with different areas. Is it possible to have multiple line colors depending on the value?
Does it work with normal views or better to switch to compose? (I still have no experience to compose framework and have a lot of respect for the work todo...).

I think, I would create a graph like this:

![GDH_Graph1](https://github.com/user-attachments/assets/6472089c-02e0-4889-b7de-176524699a95)

Sorry, it is very basic (made with paint :-) ). Not sure about the horizontal lines, better use also gray for them.

I know, you have good documentation, but is looks a little bit complicated at the moment, maybe it will be more clear, after starting development. 

Thank you very much!

### UI framework(s)

Views

**Comment by Gowsky** [comment]
Hello, @pachi81. Yes, this is possible, both in Compose and in views. It can be done via `LineCartesianLayer.LineFill`. At the moment, there are factory functions for one or two fill areas (`LineFill.single` and `LineFill.double`). Since, based on the image, you need more fill areas, a custom `LineFill` implementation will be required. However, this is straightforward — it comes down to drawing a few rectangles (five for the chart in the image). In the future, to further simplify this, we plan on adding a factory function that supports any number of fill areas.

As for the horizontal lines, you can use `HorizontalLine`.

**Comment by pachi81** [comment]
Oh, sorry for late response. I completely forgot.
At the moment, I'm using the Android Chart library, but there is no development anymore and had big troubles. So maybe I will change to this library in the future, if I have some time.
Thank you for your work!!!
